molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/filter/ClinvarMultiAllelicResultFilter.java

```diff
@@ -77,15 +82,29 @@ else if (significantAlleleIndex == 0)
 					// 1 based so we need subtract 1 from the clnAll value
 					else
 					{
+						
 						significantAlleleIndex = significantAlleleIndex - 1;
 
-						for (String annotatedEntityAltAllele : annotatedEntityAltAlleles)
+						for(int j = 0; j < annotatedEntityAltAlleles.length; j++ )
 						{
+							
 							// if annotated entity allele equals the clinvar significant allele we want it!
-							if (alts[significantAlleleIndex].equals(annotatedEntityAltAllele))
+							if (alts[significantAlleleIndex].equals(annotatedEntityAltAlleles[j]))
 							{
-								clnallValueMap.put(alts[significantAlleleIndex], clnAll[i]);
-								clnsigValueMap.put(alts[significantAlleleIndex], clnSigs[i]);
+								// if more than one clinsigs are available pair the right one with each allele
+								if (clnSigs[i].length() > 1)
+								{
```

Comment:
Code duplication?
Maybe you can create one generic method to merge these two chunks of code